[dynamo] Partially support random.Random class

yiming0416 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133037

This partially fixes the graph break issue when instantiating a `random.Random` class in Python.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

yiming0416 committed:
[dynamo] Partially support random.Random class

[ghstack-poisoned]

anijain2305 commented on torch/_dynamo/variables/user_defined.py:
```diff
@@ -489,6 +489,8 @@ def call_function(
             options = {"mutable_local": MutableLocal()}
             return variables.EnumVariable.create(self.value, args[0], options)
+        elif self.value is random.Random:
+            return RandomVariable(self.value, kwargs)
```
This is incorrect. You will have to first create a random.Random object. For example, if you have a seed here. You should 
```
if len(args) == 1 and isinstance(args[0], ConstantVariable):
	seed = args[0].value

random_object = random.Random(seed)
return RandomVariable(random_object)
```

anijain2305 reviewed (changes requested):
Left a comment

yiming0416 committed:
Update on "[dynamo] Partially support random.Random class"


This partially fixes the graph break issue when instantiating a `random.Random` class in Python.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

yiming0416 commented:
@pytorchbot merge
